tagstudio/src/core/utils/refresh_dir.py

```diff
@@ -49,24 +53,43 @@ def refresh_dir(self, lib_path: Path) -> Iterator[int]:
         self.files_not_in_library = []
         dir_file_count = 0
 
-        for path in lib_path.glob("**/*"):
-            str_path = str(path)
-            if path.is_dir():
+        for root, _, files in lib_path.walk():
+            if "$RECYCLE.BIN" in str(root).upper():
                 continue
 
-            if "$RECYCLE.BIN" in str_path or TS_FOLDER_NAME in str_path:
+            # - if directory contains file `.ts_noindex` then skip the directory
+            if constants.TS_FOLDER_NOINDEX in files:
+                logger.info("TS Ignore File found, skipping", directory=root)
+                # however check if the ignored files aren't in the library; if so, remove them
+                entries_to_remove = []
+                for file in files:
+                    file_path = root / file
+                    entry_path = file_path.relative_to(lib_path)
+                    if entry := self.library.get_path_entry(entry_path):
+                        entries_to_remove.append(entry.id)
```

Comment:
I'm not really a fan of the idea of deleting users' library data without asking for a confirmation first. Perhaps a safeguard could be a confirmation asking if the user wants to "delete x entries, with y of them having data" (assuming that more than 0 have any user data to begin with)?